### PR TITLE
Fix MP4 range requests for access control playback

### PR DESCRIPTION
**Pull request by mjh1**
Seeking in the player was broken because we weren't signaling that we support range requests. I have added an http HEAD handler for this and implemented a custom OPTIONS handler.

### CODE REVIEW

**Review by thomshutt — APPROVED** [review]

File: handlers/playback.go [hunk 1]
```diff
@@ -36,14 +37,32 @@ func PlaybackHandler() httprouter.Handle {
...
 
+		w.Header().Set("accept-ranges", "bytes")
 		w.Header().Set("content-type", response.ContentType)
+		if response.ContentLength != nil {
+			w.Header().Set("content-length", fmt.Sprintf("%d", *response.ContentLength))
+		}
+		w.Header().Set("etag", response.ETag)
+		w.WriteHeader(http.StatusOK)
+
+		if req.Method == "HEAD" {
```
Comment (victorges) [inline handlers/playback.go:48]: **nit:** 
```suggestion
		if req.Method == http.MethodHead {
```

File: api/http.go [hunk 2]
```diff
@@ -59,15 +59,16 @@ func NewCatalystAPIRouter(cli config.Cli, vodEngine *pipeline.Coordinator, bal b
...
+	playback := withLogging(
+		withCORS(
+			withGatingCheck(
+				handlers.PlaybackHandler(),
 			),
 		),
 	)
+	router.GET("/asset/hls/:playbackID/*file", playback)
+	router.HEAD("/asset/hls/:playbackID/*file", playback)
+	router.OPTIONS("/asset/hls/:playbackID/*file", withLogging(handlers.PlaybackOptionsHandler()))
```
Comment (victorges) [inline api/http.go:71]: I think we might need the `withCORS` here as well, wdyt?

File: handlers/playback.go [hunk 3]
```diff
@@ -36,14 +37,32 @@ func PlaybackHandler() httprouter.Handle {
 		}
 		defer response.Body.Close()
 
+		w.Header().Set("accept-ranges", "bytes")
```
Comment (victorges) [inline handlers/playback.go:40]: Should this be set on OPTIONS response as well?

**Review by victorges — APPROVED** [review]
nice

File: api/http.go (hunk 2 above)
Comment (mjh1) [reply api/http.go:71]: oh yep, think you're right

File: handlers/playback.go (hunk 3 above)
Comment (mjh1) [reply handlers/playback.go:40]: I don't think it's needed as it's only to check supported http methods right? but wouldn't do any harm i guess

**Pull request merged**